src/main/webapp/app/file-upload-submission/file-upload-submission.component.ts

```diff
@@ -0,0 +1,193 @@
+
+                if (this.fileUploadExercise.course) {
+                    this.complaintService.getNumberOfAllowedComplaintsInCourse(this.fileUploadExercise.course.id).subscribe((allowedComplaints: number) => {
+                        this.numberOfAllowedComplaints = allowedComplaints;
+                    });
+                }
+
+                if (data.submissions && data.submissions.length > 0) {
+                    this.submission = data.submissions[0] as FileUploadSubmission;
+                }
+
+                this.isActive =
+                    this.fileUploadExercise.dueDate === undefined || this.fileUploadExercise.dueDate === null || new Date() <= moment(this.fileUploadExercise.dueDate).toDate();
+            },
+            (error: HttpErrorResponse) => this.onError(error),
+        );
+    }
+
+    /**
+     * Uploads a submission file and submits File Upload Exercise
+     */
+    submit() {
+        const file = this.submissionFile;
+
+        if (!this.submission || !file) {
+            return;
+        }
+
+        this.erroredFile = null;
+        this.fileUploaderService.uploadFile(file, file['name'], { keepFileName: true }).then(
+            result => {
+                this.submission!.filePath = result.path;
+                this.submitExercise();
+            },
+            error => {
+                console.error('Error during file upload in uploadBackground()', error.message);
+                this.erroredFile = file;
+                this.fileInput.nativeElement.value = '';
+                this.submission!.filePath = null;
+                this.submissionFile = null;
+            },
+        );
+    }
+
+    /**
+     * Submits File Upload Exercise
+     */
+    submitExercise() {
+        const confirmSubmit = window.confirm(this.submissionConfirmationText);
+
+        if (confirmSubmit) {
+            this.submission!.submitted = true;
+            this.fileUploadSubmissionService.update(this.submission!, this.fileUploadExercise.id).subscribe(
+                response => {
+                    this.submission = response.body!;
+                    this.result = this.submission.result;
+
+                    if (this.isActive) {
+                        this.jhiAlertService.success('artemisApp.fileUploadExercise.submitSuccessful');
+                    } else {
+                        this.jhiAlertService.warning('artemisApp.fileUploadExercise.submitDeadlineMissed');
+                    }
+                },
+                err => {
+                    this.jhiAlertService.error('artemisApp.fileUploadExercise.error');
+                    this.submission!.submitted = false;
+                },
+            );
+        }
+    }
+
+    /**
+     * Sets file submission for exercise
+     * @param $event {object} Event object which contains the uploaded file
+     */
+    setFileSubmissionForExercise($event: any): void {
+        if ($event.target.files.length) {
```

Comment:
I think we also need to check the allowed file size. At least to avoid users to upload files larger than 10 MB. Maybe this could be a setting in future.

Reply:
This is already supported from jhipster. Just an important info that users can't upload more than 10MB.

Reply:
Another problem that we noticed here:
Try uploading a JAR or XLSX, it will be denied with the message that the file is too large. It seems like a security measure though.

Reply:
This is what we have on the File Resource: `Unsupported file type! Allowed file types: .png, .jpg, .svg, .pdf, .zip`. And it is also checked in the FileUploaderService. My suggestion is to not change it for now, only if we have a need for it. Allowing any file extension is a bad idea

Reply:
I think it's better now because the implementation complains that the file type is not supported (before it was always complaining about the file size).